ArcMovies/ArcMovies/ViewModels/MainPageViewModel.cs

```diff
@@ -0,0 +1,134 @@
+        public DelegateCommand<Movie> ItemAppearingCommand { get; }
+
+        private readonly INavigationService _navigationService;
+        public MainPageViewModel(INavigationService navigationService)
+        {
+            Title = "Upcoming Movies";
+            this._navigationService = navigationService;
+            Movies = new ObservableRangeCollection<Movie>();
+
+            LoadUpcomingMoviesCommand = new DelegateCommand(async () => await ExecuteLoadUpcomingMoviesCommand().ConfigureAwait(false));
+            ShowSearchMoviesCommand = new DelegateCommand(async () => await ExecuteShowSearchMoviesCommand().ConfigureAwait(false));
+            ShowMovieDetailCommand = new DelegateCommand<Movie>(async (Movie movie) => await ExecuteShowMovieDetailCommand(movie).ConfigureAwait(false));
+            ItemAppearingCommand = new DelegateCommand<Movie>(async (Movie movie) => await ExecuteItemAppearingCommand(movie).ConfigureAwait(false));
+
+            LoadUpcomingMoviesCommand.Execute();
+        }        
+
+        private async Task ExecuteLoadUpcomingMoviesCommand()
+        {
+            IsConnected = CrossConnectivity.Current.IsConnected;
+
+            if (IsBusy || !IsConnected)
+                return;
+
+            IsBusy = true;
+
+            try
+            {
+                Movies.Clear();
+                _currentPage = 1;
+                await LoadMoviesAsync(_currentPage, Enums.MovieCategory.Upcoming).ConfigureAwait(false);
+            }
+            finally
+            {
+                IsBusy = false;
+            }
+        }
+
+        private async Task ExecuteShowSearchMoviesCommand()
+        {            
+            await _navigationService.NavigateAsync("SearchMoviesPage").ConfigureAwait(false);
+        }
+
+        private async Task ExecuteShowMovieDetailCommand(Movie movie)
+        {            
+            var parameters = new NavigationParameters();
+            parameters.Add(nameof(movie), movie);
+            await _navigationService.NavigateAsync("MovieDetailPage", parameters).ConfigureAwait(false);
+        }
+
+        private async Task ExecuteItemAppearingCommand(Movie movie)
+        {
+            int itemLoadNextItem = 2;
+            int viewCellIndex = Movies.IndexOf(movie);
+            if (Movies.Count - itemLoadNextItem <= viewCellIndex)
+            {                
+                await NextPageUpcomingMoviesAsync().ConfigureAwait(false);
+            }
+        }
+
+        private async Task LoadMoviesAsync(int page, Enums.MovieCategory movieCategory)
+        {
+            try
+            {
+                var continueOnCapturedContext = Device.RuntimePlatform == Device.Windows;
+
+                _genres = _genres ?? await ApiService.GetGenresAsync().ConfigureAwait(continueOnCapturedContext);
+                var searchMovie = await ApiService.GetMoviesByCategoryAsync(page, movieCategory).ConfigureAwait(continueOnCapturedContext);
```

Comment:
Since both these calls are not dependent upon each other you could've triggered them both at the same time and wait for both of them to finish.